eclass/tests/Makefile

```diff
@@ -0,0 +1,27 @@
+SH_FILES := $(wildcard *.sh)
+TEST_FILES := $(filter-out tests-common.sh, $(SH_FILES))
+TEST_OK_FILES := $(patsubst %.sh, .%.sh.ok,$ $(TEST_FILES))
+
+# We cache a successful test result if the testfile itself did not
+# change (%.sh) and the contents of the eclass/ directory did not
+# change (.eclasssum).
+.%.sh.ok: %.sh .eclasssum
+	./$<
+	touch $@
+
+.PHONY: test
+test: $(TEST_OK_FILES)
+
+.PHONY: force
+.ONESHELL:
+.eclasssum: SHELL = /bin/bash
+.eclasssum: force
+	set -eu
+	find .. -maxdepth 1 -type f -name "*.eclass" |\
+		xargs stat --format="%n %y" |\
```

Comment:
Why `xargs` over `-exec stat --format="%n %y" \{} \+` ?

Reply:
Also, maybe %Y for mtime rather than %y format.

Reply:
Thanks, will use `execdir` instead of `xargs`.
`%Y`'s precision is "only" seconds, whereas `%y` has nanoseconds.